src/lib/mockserver.js

```diff
@@ -139,6 +118,47 @@ class MockServer {
             }
         }
     }
+
+    /**
+     * Spawn docker logs until mocks are set up
+     */
+    async _dockerSpawnLogs() {
+        debug(`> docker logs -f ${this.container}`);
+        const proc = spawn('docker', ['logs', '-f', this.container]);
+        let isRunning;
+
+        proc.stdout.on('data', function(data) {
+            const stdout = data.toString();
+            if (stdout && stdout.includes('started on ports: [80, 443]')) {
+                debug("Stdout", stdout);
+                isRunning = true;
+            }
+        });
+        proc.stderr.on('data', function(data) {
+            process.stderr.write(data.toString());
+        });
+
+        const kill = proc.kill;
+        proc.kill = function() {
+            // ensure logging stops immediately
+            proc.stdout.removeAllListeners("data");
+            proc.stderr.removeAllListeners("data");
+            kill.apply(proc);
+        };
```

Comment:
I don't think you need that here. Over [here](https://github.com/adobe/aio-cli-plugin-asset-compute/blob/master/src/lib/actionrunner.js#L329-L336) this was needed for something that does not apply here.

It was writing the docker logs to stdout/err of the process itself. The kill wasn't immediate, so sometimes it continued writing some stuff on the console for a few lines, which was messing up the output there. Here we don't write anything, just care about watching for "started on ports" in the output, so it can be dropped.